app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -41,9 +41,9 @@ def redirect_or_render(auth)
     end
 
     def pre_auth
-      @pre_auth ||= OAuth::PreAuthorization.new(Doorkeeper.configuration,
-                                                server.client_via_uid,
-                                                params)
+      @pre_auth ||= Doorkeeper.configuration.preauth_class.new(Doorkeeper.configuration,
```

Comment:
Yes I don't mean the white space, which can be better or worse but at last it's superficial. I mean that code is harder to read, and I don't know if it's because of how we are tackling the problem, or because of naming, or because we are getting too many configuration parameters. It looks like a smell of something not going well somewhere. Next week I will have more time to think about this and come up with more concrete answers.